src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -67,8 +66,8 @@ public void AddAllTo(HashSet<Checksum> checksums)
         IReadOnlyList<T> values,
         ChecksumCollection checksums,
         HashSet<Checksum> searchingChecksumsLeft,
-        Dictionary<Checksum, object?> result,
-        CancellationToken cancellationToken)
+        Dictionary<Checksum, object> result,
+        CancellationToken cancellationToken) where T : class
     {
         Contract.ThrowIfFalse(values.Count == checksums.Children.Length);
```

Comment:
Github doesn't seem to allow comments in sections of the file that weren't changed? For the two comments that were unclear, they were both in the loop below and were a bit nit-picky, so I don't care too much if they aren't taken:

1) No need to check the cancellation token in the loop, nothing expensive going on in there.
2) The .Count check could just be done inside the .Remove condition as it's status can't change unless the Remove condition succeeds.

Reply:
Fair. I can revise!